agent/lib/docker/container.rb

```diff
@@ -79,6 +86,53 @@ def env_hash
       @env_hash
     end
 
+    # @return [Hash]
+    def networks
+      cached_json.dig('NetworkSettings', 'Networks')
+    end
+
+    def has_network?(network)
+      self.networks.has_key?(network)
+    end
+
+    # IP address of container within named network, or nil.
+    #
+    # @param network [String] Docker network name
+    # @return [String, nil]
+    def network_ip(network)
+      cached_json.dig('NetworkSettings', 'Networks', network, 'IPAddress')
+    end
+
+    # CIDR address of container within named network, or nil.
+    #
+    # @param network [String] Docker network name
+    # @return [String, nil]
+    def network_cidr(network)
+      if network = cached_json.dig('NetworkSettings', 'Networks', network) && network['IPAddress']
+        return "#{network['IPAddress']}/#{network['IPPrefixLen']}"
+      else
+        return nil
+      end
+    end
+
+    # Container IP address within the overlay network.
+    # Will be missing/nil if container is not attached to the overlay network.
+    #
+    # Plain IP address without any CIDR suffix.
+    #
+    # @return [String, NilClass]
+    def overlay_ip
```

Comment:
This is used in three places:

* weave attach
* LB registrator
* health check worker

Each of those will need some changes once we have multiple networks.